### PR TITLE
chore: Convert SentrySessionReplay to Swift

### PR DESCRIPTION
**Pull request by brustolin**
## :scroll: Description

Converted SentrySessionReplay to Swift and started using a delegate approach to remove SentrySDK and SentryHub dependency from `SentrySessionReplay`

## :green_heart: How did you test it?

Unit tests

_#skip-changelog_

### CODE REVIEW

File: Sources/Sentry/include/SentryPrivate.h [hunk 1]
```diff
@@ -1,16 +1,17 @@
 // Sentry internal headers that are needed for swift code
-
 #import "SentryBaggage.h"
 #import "SentryDispatchQueueWrapper.h"
 #import "SentryNSDataUtils.h"
 #import "SentryRandom.h"
-#import "SentryReplayType.h"
+// #import "SentryReplayType.h"
```
Comment (brustolin) [inline Sources/Sentry/include/SentryPrivate.h:6]:
```suggestion
```

File: Sources/Swift/Integrations/SessionReplay/SentryReplayEvent.swift [hunk 2]
```diff
@@ -0,0 +1,48 @@
+@_implementationOnly import _SentryPrivate
+import Foundation
+
+@objcMembers
+class SentryReplayEvent: Event {
```
Comment (krystofwoldrich) [inline Sources/Swift/Integrations/SessionReplay/SentryReplayEvent.swift:5]: Compared to the obj-c impl trace_ids and error_ids are missing, were they not used?

File: Sources/Swift/Integrations/SessionReplay/SentrySessionReplay.swift [hunk 3]
```diff
@@ -0,0 +1,286 @@
+import Foundation
+#if (os(iOS) || os(tvOS)) && !SENTRY_NO_UIKIT
```
Comment (krystofwoldrich) [inline Sources/Swift/Integrations/SessionReplay/SentrySessionReplay.swift:2]: Can we keep using `#if SENTRY_TARGET_REPLAY_SUPPORTED`?

Comment (brustolin) [reply Sources/Swift/Integrations/SessionReplay/SentrySessionReplay.swift:2]: Swift doesn't have this Objc macros

Comment (vaind) [reply Sources/Swift/Integrations/SessionReplay/SentrySessionReplay.swift:2]: are you 100% sure? AFAIK simple macros work fine

**Review by krystofwoldrich — APPROVED** [review]
Looks good! More Swift!

I had mostly question to understand, otherwise good.

File: Sources/Sentry/SentrySessionReplayIntegration.m [hunk 4]
```diff
@@ -186,7 +187,7 @@ - (void)sentrySessionStarted:(SentrySession *)session
 
 - (void)captureReplay
 {
-    [self.sessionReplay captureReplay];
+    //[self.sessionReplay captureReplay];
```
Comment (vaind) [inline Sources/Sentry/SentrySessionReplayIntegration.m:190]: Was this meant to be commented out in the final version? It stopped hybrid SDKs from sending replays.

Comment (vaind) [reply Sources/Sentry/SentrySessionReplayIntegration.m:190]: forgot to tag @brustolin 

cc/fyi @krystofwoldrich if you notice stuff doesn't work for you in RN

**Pull request merged**